### ISSUE TITLE
lint `&ref x`  patterns, they could be just `x`

### ISSUE THREAD

**Issue by oli-obk** [rust-lang-nursery/rust-clippy#1434]
Example:

```rust
    let mut v = Vec::<String>::new();
    let _ = v.iter_mut().filter(|&ref a| a.is_empty());
```

could just be `|a| a.is_empty()`

**Comment by CBenoit** [comment]
Hello, I would like to start contributing Clippy with this issue!

**Comment by oli-obk** [comment]
great! if you have any questions, feel free to ask here or open an (unfinished) PR to discuss.

**Comment by oli-obk** [comment]
You should probably look at the `needless_ref` lint and report the new lint somewhere inside the code that reports `needless_ref`

**Comment by CBenoit** [comment]
Thanks! I looked for a `needless_ref` in the `rust-clippy` folder but didn't found it. Did you refer to `needless_borrow` or am I missing something?

**Comment by oli-obk** [comment]
That's the one, got confused about the names.

**Comment by Phlosioneer** [comment]
Triage here: This appears to be closed by #1536?
